apps/dashboard/app/views/files/index.html.erb

```diff
@@ -380,33 +378,60 @@ $('#new-dir-btn').on("click", () => {
   .then((filename) => newDirectory(filename));
 });
 
-function downloadFile(file){
+function downloadFile(file) {
```

Comment:
How about two separate functions: downloadFile and downloadDirectory?

The downloadDirectory function would have the download directory request, and then if that succeeds, it would call downloadFile.

That would move the control flow down to the `$('#download-btn').on("click", () => {` but that's probably a more appropriate place.